aiida/cmdline/utils/daemon.py

```diff
@@ -126,8 +126,13 @@ class StartCircusNotFound(Exception):
     if pid is not None:
         try:
             process = psutil.Process(pid)
+            # the PID got recycled, but a different process
             if _START_CIRCUS_COMMAND not in process.cmdline():
                 raise StartCircusNotFound()  # Also this is a case in which the process is not there anymore
+
+            # the PID got recycled, but for a daemon of someone else
+            if process.username() != psutil.Process().username():  # compare against the username of this interpreter
+                raise StartCircusNotFound()
```

Comment:
a small piece of advice,  since the exception includes a new case the name should be more generalized(maybe `IncorrectCircusProcess`?). As well as the class docstring of Exception in the few lines above (121). @dev-zero

Reply:
Not really: the `StartCircusNotFound` is raised when the previous circus process could not be found. This is the same thing in the newly added check: a process with the same PID and `cmdline` was found, but for a different user, hence it's not the previously remembered circus process, hence the conclusion is still "not found in the ps command". In other terms: just the conditions under which a process is considered to be the previously remembered one are tightened a bit.

Reply:
I see, thanks @dev-zero. Anyway, this is a useful fix, I remember I encountered the case where I have to delete staled circus pid file manually.